library/common/types/c_types.h

```diff
@@ -113,6 +113,12 @@ envoy_data copy_envoy_data(size_t length, const uint8_t* src_bytes);
 // For example when sending a headers-only request.
 extern const envoy_data envoy_nodata;
 
+// Convenience constant to return when data allocation fails.
+extern const envoy_data envoy_nulldata;
+
+// Convenience constant to return when data allocation fails.
+extern const envoy_headers envoy_nullheaders;
+
```

Comment:
And so I think these can be removed (since we'll use structs with a non-zero length but with nullptr bytes).